test/kubemark/start-kubemark.sh

```diff
@@ -426,6 +420,32 @@ function start_hollow_nodes_scaleout {
   done
 }
 
+# setup_proxy setups the proxy service for the scale-out deployment
+# and creates kubeconfig with the proxy service IP and port
+function setup_proxy {
+  local -r tp1_ip=$(grep server "${TP_KUBECONFIG}-1" | awk -F "/" '{print $3}')
+  export TPIP="${tp1_ip}"
+  for (( tp_num=2; tp_num<=${SCALEOUT_TP_COUNT}; tp_num++ ))
+  do
+    tp_ip=$(grep server "${TP_KUBECONFIG}-${tp_num}" | awk -F "/" '{print $3}')
+    export TPIP=${TPIP},"${tp_ip}"
+  done
+
+  # currently proxy only supports 1 RP
+  rp1_ip=$(grep server "${RP_KUBECONFIG}-1" | awk -F "/" '{print $3}')
+  export RPIP="${rp1_ip}"
+
+  export KUBERNETES_SCALEOUT_PROXY=true
+  export KUBE_BEARER_TOKEN=${SHARED_APISERVER_TOKEN}
+  export SCALEOUT_PROXY_NAME="${KUBE_GCE_INSTANCE_PREFIX}-proxy"
+  export PROXY_KUBECONFIG="${RESOURCE_DIRECTORY}/kubeconfig.kubemark-proxy"
+  export KUBERNETES_SCALEOUT_PROXY_APP=${KUBERNETES_SCALEOUT_PROXY_APP:-haproxy}
+  export PROXY_CONFIG_FILE=${PROXY_CONFIG_FILE:-"haproxy.cfg"}
+  export PROXY_CONFIG_FILE_TMP="${RESOURCE_DIRECTORY}/${PROXY_CONFIG_FILE}.tmp"
+  create-arktos-proxy
+  export KUBERNETES_SCALEOUT_PROXY=false
```

Comment:
what is the reason for this var being reset?

Reply:
KUBERNETES_SCALEOUT_PROXY is reset to its init value just to ensure the code below this point get the correct state.